tests/test_live_events.py

```diff
@@ -65,6 +66,9 @@ def setUp(self):
         self.area_grid = Area("Grid", children=[self.area_house1, self.area_house2],
                               config=self.config)
 
+    def tearDown(self) -> None:
+        GlobalConfig.sim_duration = duration(days=1)
```

Comment:
Now I see why people don't like singletons :O Basically, here we have to restore the initial value of `GlobalConfig.sim_duration` even though we never directly changed: it was changed somewhere inside one of the classes we're using, but this effect is not reversed at the end of the test. If we start having too many singletons (buried deep in the classes hierarchy) this would become a nightmare.

Reply:
That's one of the long list of reasons that singletons are more or less an antipattern indeed. In general, there are 2 cases for a class to be allowed to be a singleton: 
1. configuration shared across the application (this case)
2. database or network connections, since these are singletons by nature

As you can see, even though we have a typical usecase for a singleton here, it is not without problems, since the singleton has become more or less a set of global variables in the unit tests context. For the production code we do not have this problem because the GlobalConfig is only instantiated once when the simulation is started and never mutated again, still for the tests it is very annoying